pkg/autoscaler/fake/fake_metric_client.go

```diff
@@ -0,0 +1,60 @@
+/*
```

Comment:
This is great! Do you mind putting the fake metric client in it's own PR? It neatly cleans up everything but it widens the diff of this change by a lot. I think that'd help reviewability.

Reply:
#4991